presto-orc/src/main/java/com/facebook/presto/orc/metadata/DwrfStripeCacheWriter.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.orc.metadata;
+
+import com.facebook.presto.common.io.DataOutput;
+import com.google.common.collect.ImmutableList;
+import io.airlift.slice.DynamicSliceOutput;
+import io.airlift.units.DataSize;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.lang.Math.toIntExact;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Accumulates data for DWRF stripe cache using given mode and max size.
+ */
+public class DwrfStripeCacheWriter
+{
+    private final DwrfStripeCacheMode mode;
+    private final DynamicSliceOutput cache;
+    private int size;
+    private final long maxSize;
+    private boolean canWriteMore = true;
+    private final List<Integer> offsets = new ArrayList<>();
+
+    public DwrfStripeCacheWriter(DwrfStripeCacheMode mode, DataSize maxSize)
+    {
+        // make sure max size is in the int range
+        this.maxSize = toIntExact(requireNonNull(maxSize, "maxSize is null").toBytes());
+        this.mode = requireNonNull(mode, "mode is null");
+        this.cache = new DynamicSliceOutput(64);
```

Comment:
Is 64 bytes too small of a size to start with ?

Reply:
I'm sorry, I don't get the Q here. Do you suggest to use different size for various modes like NA and non-NA? DynamicSliceOutput cannot have 0 size, so I used some random small enough value.

Reply:
I was wondering why the number 64, as 64 looks smaller for the stripe cache.

Reply:
64 is nothing in case of NA and easily scalable in case of non-NA mode. If you have a particular number in mind I can change it. Let's say a few KBs, like 32KB, can fit cache for files with a small number of stripes and small schema.

Reply:
Here is where the 64 came from:
```java
    public Slice writePostscript(int footerLength, int metadataLength, CompressionKind compression, int compressionBlockSize)
            throws IOException
    {
        // postscript is not compressed
        DynamicSliceOutput output = new DynamicSliceOutput(64);
        metadataWriter.writePostscript(output, footerLength, metadataLength, compression, compressionBlockSize, dwrfStripeCacheData);
        return output.slice();
    }
```